### ISSUE TITLE
[REVIEW]: Excimontec v1.0: An Open-Source Software Tool for Kinetic Monte Carlo Simulations of Organic Electronic Devices

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2307]
**Submitting author:** @MikeHeiber (<a href="http://orcid.org/0000-0002-1567-5663">Michael Heiber</a>)
**Repository:** <a href="https://github.com/MikeHeiber/Excimontec" target ="_blank">https://github.com/MikeHeiber/Excimontec</a>
**Version:** v1.0.0
**Editor:** @pdebuyl
**Reviewers:** @mdavezac, @Luthaf
**Archive:** <a href="https://doi.org/10.5281/zenodo.4008122" target="_blank">10.5281/zenodo.4008122</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/1db972a24b6d9a9c2f7ce8658e8a1778/status.svg)](https://joss.theoj.org/papers/1db972a24b6d9a9c2f7ce8658e8a1778)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/1db972a24b6d9a9c2f7ce8658e8a1778"><img src="https://joss.theoj.org/papers/1db972a24b6d9a9c2f7ce8658e8a1778/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/1db972a24b6d9a9c2f7ce8658e8a1778/status.svg)](https://joss.theoj.org/papers/1db972a24b6d9a9c2f7ce8658e8a1778)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/MikeHeiber/Excimontec" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@mdavezac, @Luthaf, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @pdebuyl know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/MikeHeiber/Excimontec">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@MikeHeiber) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/MikeHeiber/Excimontec">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@MikeHeiber) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @mdavezac it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1002/pssb.201350339 is OK
- 10.1088/1361-6633/80/2/026502 is OK
- 10.21105/joss.01072 is OK
- 10.1016/B978-0-08-102284-9.00010-3 is OK
- 10.21105/joss.01168 is OK
- 10.1002/aelm.201800821 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02307/joss.02307/10.21105.joss.02307.pdf)

**Comment by pdebuyl** [comment]
@whedon add @liangtianumich as reviewer

**Comment by whedon** [comment]
OK, @liangtianumich is now a reviewer

**Comment by pdebuyl** [comment]
@mdavezac, @liangtianumich , make sure to accept the invitation to the reviewers group and to have a look at the reviewer guidelines.

The review process will happen in this issue page, so questions to the author or to me can be added as comments here.

**Comment by mdavezac** [comment]
@pdebuyl, I missed the invitation to the reviewer group and it time out. Could you please send it back?

**Comment by pdebuyl** [comment]
@arfon is it possible for me to restart an invitation to the reviewer group?

**Comment by arfon** [comment]
It is, you can do `@whedon re-invite @reviewer as reviewer`

**Comment by pdebuyl** [comment]
@whedon re-invite @mdavezac  as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@mdavezac  please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by pdebuyl** [comment]
@arfon thanks

**Comment by mdavezac** [comment]
## Compilation

The procedure completed almost without a hitch. The issue is that it requires the users to set up their environment with a CXX variable. This may be a standard for  a developer, not so much for user. This requirement is missing from the installation guide.

What's weird though is that the environment variable is not actually used to identify the compiler, since the compiler is **hard-coded** to mpicxx.  It's merely a surprising way to specify a set of CXXFLAGS. However, even though CXX is just a switch, it cannot be set in the expected way for  a makefile, e.g. `make CXX=xxx`. It has to be done via an environment variable. And the environment variable cannot be something like `g++-8.0`. Hence, the setup will fail in any number of standard situations where CXX points to an actual versioned executable. 

Among other environments, this makefile will fail on crays, where mpi compiler wrappers have less standard names.

## Installation

There is no installation procedure per se. However, the main product is a relocatable executable.

## Recommendation

At the low end, it would be sufficient to explain the non-standard usage of CXX in the user-guide and that the executable can be relocated to another location of the user's choosing. It may be  good idea to replace CXX with a suitably named makefile variable with a suitable default.

However, I would **strongly** recommend ditching hand-rolled makefiles in favor of [meson](https://mesonbuild.com) or [cmake](https://www.cmake.org). The advantages will include: (i) out of tree builds, (ii) large set of compilers, (iii) standard installation procedure to /usr/bin, (v) installation of libKMC.a (or .so),  (v) build specification shorter and less error prone than a makefile.
Furthermore, it would give the author the opportunity to replace the git submodules with either [meson's dependency wrapper](https://mesonbuild.com/Wrap-dependency-system-manual.html) or [cmake's external project](https://cmake.org/cmake/help/latest/module/ExternalProject.html). Submodules are a source of confusion even for expert git users, especially in a multi-developer environment.

**Comment by mdavezac** [comment]
Software paper: Travis provide continuous integration. [Integration testing](https://en.wikipedia.org/wiki/Integration_testing) is something else still. Indeed some of the unit-tests in the code could be termed integration tests, e.g. ExcitonDiffusionTests.

**Comment by pdebuyl** [comment]
Thank you @mdavezac for the review!

@MikeHeiber you can already reply to @mdavezac , you don't have to wait for @liangtang 's review.

The process at JOSS is to discuss the comments as a dialogue on this github issue page.

**Comment by pdebuyl** [comment]
@amandadumi would you review the article "Excimontec v1.0: An Open-Source Software Tool for Kinetic Monte Carlo Simulations of Organic Electronic Devices" for the Journal of Open Source Software?

**Comment by pdebuyl** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a  dry run of accepting the paper and depositing with Crossref
@whedon accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by pdebuyl** [comment]
@whedon remove @liangtianumich as reviewer

**Comment by whedon** [comment]
OK, @liangtianumich is no longer a reviewer

**Comment by pdebuyl** [comment]
@whedon add @mdavezac @luthaf as reviewers

**Comment by whedon** [comment]
OK, @mdavezac @luthaf is now a reviewer

**Comment by pdebuyl** [comment]
Thank you @Luthaf for accpeting the invitation.

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let me know.

**Comment by danielskatz** [comment]
note that command did not do what you intended
you now have 2 reviewers
1 - @mdavezac 
2 - @mdavezac @Luthaf
whedon only can handle one at a time.  You should remove "@mdavezac @Luthaf" and then add "@Luthaf"

**Comment by pdebuyl** [comment]
Thank you @danielskatz

**Comment by pdebuyl** [comment]
@whedon remove @mdavezac @Luthaf as reviewers

**Comment by whedon** [comment]
OK, @mdavezac @Luthaf is no longer a reviewer

**Comment by pdebuyl** [comment]
@whedon add @Luthaf as reviewer

**Comment by whedon** [comment]
OK, @Luthaf is now a reviewer

**Comment by danielskatz** [comment]
@whedon re-invite @Luthaf as reviewer

(this may also be needed - not sure, but it won't hurt to try)

**Comment by whedon** [comment]
The reviewer already has a pending invite.

@luthaf please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by Luthaf** [comment]
I accepted the invite, but I can't check the boxes in the initial message. Any idea on what might be wrong?

**Comment by pdebuyl** [comment]
On Thu, Jul 02, 2020 at 06:39:35AM -0700, Guillaume Fraux wrote:
> I accepted the invite, but I can't check the boxes in the initial message. Any idea on what might be wrong?

I added you in the assigned persons for the issue page, can you try again?

**Comment by Luthaf** [comment]
I reloaded the page in the mean time, and it worked (hence I deleted the comment visible in @pdebuyl email answer). Everything is fine now!

**Comment by pdebuyl** [comment]
Good to know, thank you.

**Comment by Luthaf** [comment]
I agree with @mdavezac comment regarding compilation (https://github.com/openjournals/joss-reviews/issues/2307#issuecomment-648739993); and also recommend using CMake/Autotools/Meson to build Excimontec (my personal preference being for CMake). I don't mind the use of git submodules, if they are well integrated with the build system everything should be fine.

Since changing the build system can be a bit of work (although very much worth it); you could at least fix the issue around CXX by using `mpicxx -showme` to get the actual compiler name, and potentially running `$(compiler) --version` to detect gcc/clang/pgi/cray compilers/...

**Comment by pdebuyl** [comment]
@MikeHeiber Can you proceed to reply to the current reviews?

**Comment by MikeHeiber** [comment]
Thank you @mdavezac and @Luthaf for the feedback about the build system Excimontec is currently using.  While it might not be the most ideal solution, building the tool with GNU Make is fairly simple and it works.  I've compiled and run Excimontec using Make on a number of different computing clusters at different universities and national labs, and I've never had major issues. There's also a handful of current users around the world using the tool, and they also have not reported issues with the current build system. For this reason, it is not a high priority to update to a more sophisticated build system at this time. There is an issue filed to switch to cmake (https://github.com/MikeHeiber/Excimontec/issues/25), and I'll be monitoring users' interest in this feature as time goes on. Currently, there's mostly only users and only myself as the main developer, so feature prioritization is really critical. I tend to favor working on features that improve functionality whenever possible to continue enabling the cutting edge research being done by users. 

The way the compilation works right now is not a strange usage of the CXX environment variable.  GNU Make sets the CXX variable based on the compiler that it is set up to use. I have never needed to manually set the CXX variable before compilation. The makefile is simply detecting the compiler type that the user has set up on their system to set the compiler flags appropriately. This might not work if the user has a bunch of different compiler versions set up on their system and wants to specifically use a particular version, but in this more advanced case the super user that is interested in doing this can also simply modify the makefile to conform to their system. However, I will look into @Luthaf 's suggestions about potentially better ways to determine the compiler being used under the hood of the mpicxx command and adding support for the clang compiler.

**Comment by amandadumi** [comment]
@pdebuyl Hello, if another reviewer is still needed, I would be able and interested! Sorry for the delayed response.

**Comment by pdebuyl** [comment]
Hi @amandadumi thanks for getting back to me. For this paper it is not necessary but I'll make sure to think of you for a future review!

**Comment by mdavezac** [comment]
@MikeHeiber, I'm not sure what to say. Here are two random reviewers and both found that the build did not go without a hitch... You may be overestimating the time it would take to convert to a modern build system. For a project this size and with so few dependencies, it would not be much more time then re-inventing compiler/mpi detection in GNU Make. In any case, are you planning to **investigate this for this submission**? Otherwise, I would recommend explaining in the manual how CXX is used, or at least list the values it can take if set. 

Also, I would appreciate if you could take this comment into account: https://github.com/openjournals/joss-reviews/issues/2307#issuecomment-648748916

**Comment by pdebuyl** [comment]
Hi all,

At JOSS, we aim at improving the quality of the submitted software. However, mandating a specific build system (or otherwise specific features) is not part of our review. Besides, GNU Make is widely used and supported.

What is part of the review criteria is that the installation instructions are functional and properly documented, and that is what the review should focus on as far as the build system is concerned.

**Comment by pdebuyl** [comment]
@MikeHeiber it is often the case that reviewers wait for update on the first round of review before proceeding to the rest of the review. Please let us know when you have taken the comments into account.

**Comment by MikeHeiber** [comment]
To improve the build process, I've updated how the makefile detects the compiler that the user has installed to run with their MPI installation.  Similar to what @Luthaf suggest, the default makefile now uses the output of the `mpicxx -show` command to determine the compiler that is installed and sets the compiler flags accordingly.  This should be more robust than using the CXX environment variable as was being done previously.  I've also added clang support by adding it to the makefile compiler detection list and also by including it in my TravisCI build matrix. The latest version can be seen here:
https://github.com/MikeHeiber/Excimontec/tree/joss-paper-update

While I am working on the other improvements, such as some additions to the User Manual, please finish the other parts of the review. There should hopefully be nothing with the build process that is blocking further review progress.

**Comment by MikeHeiber** [comment]
I've made another update based on a suggestion by @Luthaf and the makefile should now be able to handle situations where the compiler underlying mpicxx is a versioned g++ executable.  For example, g++-9 will still be recognized as a g++ family compiler and the compiler flags will be set accordingly.

**Comment by pdebuyl** [comment]
Thank you @MikeHeiber for the updates.

@mdavezac @Luthaf can you give feedback on the updated installation procedure?

**Comment by Luthaf** [comment]
The build is still failing for me with clang compilers: https://github.com/MikeHeiber/Excimontec/issues/107#issuecomment-665568938

I also found a problem with character encoding in result files https://github.com/MikeHeiber/Excimontec/issues/110, although I do not consider it to be a blocking point for this review.

---

Regarding the paper itself, I am unable to comment on 

> **State of the field**: Do the authors describe how this software compares to other commonly-used packages?

since I've never used or heard of KMC simulation before. 

The paper is fairly short, and building on top two other JOSS papers by the same author: https://joss.theoj.org/papers/10.21105/joss.01072 and https://joss.theoj.org/papers/10.21105/joss.01168. It might benefit from a short description of the core KMC simulation idea; as well as how it is applied to organic electronic devices.

**Comment by MikeHeiber** [comment]
I've made a significant update to the user manual and paper itself. As before, the latest version can be seen here:
https://github.com/MikeHeiber/Excimontec/tree/joss-paper-update

To the paper, I've added additional information about what KMC simulations can be used for and specifically what types of simulation tests come with v1.0.  I also direct readers interested in more details to the user manual, which provides far more detail than is appropriate for the code paper. I've also specified that TravisCI is doing continuous integration tests instead of simply calling them integration tests.

The user manual has been significantly reorganized and updated to show and explain the parameters within each section of the parameter file and how they related to underlying physical mechanisms that are implemented. I've also added information about the parameter file format and syntax near the beginning. To address concerns with installation instructions, I've added commands to install the GCC compiler and GNU Make.

Clang compiler support has been added and should work.  It's currently being testing and is working on Ubuntu in the TravisCI build matrix. Character encoding issues in the results file will be tackled at a later date.

This should hopefully address all major concerns/issues so that we can move things forward. Thank you to the reviewers for their detailed assessment of everything and helping improve this tool's accessibility with new users.

**Comment by pdebuyl** [comment]
Thank you @MikeHeiber for the reply.

@mdavezac and @Luthaf , the process is back in your hands.

**Comment by Luthaf** [comment]
I'm currently on holidays until the 24th, I'll see if I can find some time before that but I can't make any promise

**Comment by MikeHeiber** [comment]
Can we please wrap up this review process? I have two other submitted manuscripts waiting on this. I'd really like to cite this software tool as my colleagues and I have done some cool science with it.

**Comment by Luthaf** [comment]
All of my concerns have been addressed, and I can recommend the publication of this paper in JOSS!

Regarding potential future improvements, I encourage @MikeHeiber to think about modifying the build system for easier integration with different systems (other OS/compilers/CPU architechture/etc.); and continue improving the user manual & documentation to empower both new and power users of his software.

**Comment by mdavezac** [comment]
I'm also happy to recommend the publication of the paper in JOSS!

Just a last few comments and recommendation:

- the updated build system did work. However I would recommend adding a few sentences to the installation instructions for non-standard compilers (e.g. run with COMPILER and/or FLAGS set to something) or non-standard compiler names. Such things are not unheard of on HPC platforms. Of course, there is always the option of moving to cmake or similar ;)
- the user manual is pleasingly extensive and informative. It would be even better if it where on the web in html form. Furthermore, it is better at describing what the software does than how to use it in practice. Hence it would be more valuable if the scripts to create the data (and the graphs) therein were more fully available and usable (i.e. not in tar files and with hard-coded paths to external data).
- the input file format is not great, especially considering that parameter sweeps are a likely usage scenario. I expect that the format does not scale well and requires reconciling with each version of the software. I would recommend choosing a standard, flexible, file format, e.g. [TOML](https://github.com/toml-lang/toml), [yaml](https://yaml.org/), or even XML.

**Comment by pdebuyl** [comment]
@Luthaf @mdavezac thank you for the reviews!

@MikeHeiber I will go over the paper this week for the rest of the editorial process.

**Comment by MikeHeiber** [comment]
Thank you @Luthaf and @mdavezac again for your feedback.  I have filed several issues capturing your final suggestions and will look to implement these additional improvements in the future. 

All updates derived from this review process are now merged into master, and I have released v1.0.0 of the software.

@pdebuyl @arfon I have another manuscript in the proofing stage waiting for this process to finish so that I can properly cite this JOSS software paper.  If there is anything that you can do to move this manuscript forward as soon as possible, it would be greatly appreciated.

**Comment by MikeHeiber** [comment]
@whedon set 10.5281/zenodo.4008122 as archive

**Comment by whedon** [comment]
I'm sorry @MikeHeiber, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by MikeHeiber** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02307/joss.02307/10.21105.joss.02307.pdf)

**Comment by MikeHeiber** [comment]
@whedon generate pdf from branch joss-proofing

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-proofing. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02307/joss.02307/10.21105.joss.02307.pdf)

**Comment by arfon** [comment]
> @pdebuyl @arfon I have another manuscript in the proofing stage waiting for this process to finish so that I can properly cite this JOSS software paper. If there is anything that you can do to move this manuscript forward as soon as possible, it would be greatly appreciated.

We need @pdebuyl to make his last review before handing this off to the EiC on rotation. It sounds like this should happen in the next few days.

**Comment by pdebuyl** [comment]
@whedon set https://doi.org/10.5281/zenodo.4008122 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4008122" target="_blank">10.5281/zenodo.4008122</a> is the archive.

**Comment by pdebuyl** [comment]
@whedon set v1.0.0 as version

**Comment by whedon** [comment]
OK. v1.0.0 is the version.

**Comment by pdebuyl** [comment]
@MikeHeiber what is the branch to use, now and for final publication?

**Comment by MikeHeiber** [comment]
@pdebuyl The latest version of the paper is on the joss-proofing branch.  I am finished with my edits, so if you don't have any other required edits, you can create the final publication using this branch.  I will then merge it into master once the paper it published.

**Comment by pdebuyl** [comment]
@whedon generate pdf from branch joss-proofing

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-proofing. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02307/joss.02307/10.21105.joss.02307.pdf)

**Comment by pdebuyl** [comment]
@whedon accept from branch joss-proofing

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1002/pssb.201350339 is OK
- 10.1088/1361-6633/80/2/026502 is OK
- 10.21105/joss.01072 is OK
- 10.1016/B978-0-08-102284-9.00010-3 is OK
- 10.21105/joss.01168 is OK
- 10.5281/zenodo.4008122 is OK
- 10.1002/aelm.201800821 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1692

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1692, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss-proofing 
```

**Comment by pdebuyl** [comment]
@MikeHeiber this is a good time to merge if you wish. Else, the process carries on with the other branch. One of the editors-in-chief will take over the rest of the process.

**Comment by pdebuyl** [comment]
ping @openjournals/joss-eics

**Comment by kthyng** [comment]
Hi @pdebuyl I saw this but thought I'd give @MikeHeiber a chance to do the merging before I moved ahead. Should I just go ahead now?

**Comment by pdebuyl** [comment]
Hi @kthyng thanks for the reply. The merge of the custom branch should be a "fast-forward" merge only and the software release won't change. Please go ahead with the custom branch then, @MikeHeiber is waiting to cite this paper in others, so it is better to proceed.

**Comment by MikeHeiber** [comment]
@kthyng Please go ahead and publish the paper in it's current form.  I'll merge the joss-proofing branch in afterwards and also update the branch to include the reference information.

**Comment by kthyng** [comment]
Hi @MikeHeiber! I'll help finish up the details for your publication. Please change the metadata for your zenodo archive to exactly match your JOSS paper (title and authors).

**Comment by kthyng** [comment]
I just read through your paper @MikeHeiber. It looks fine, except for some reason all the references were listed after the period of the sentence they were used in. I changed this in [this PR](https://github.com/MikeHeiber/Excimontec/pull/117). Please look through and see if you want to merge.

**Comment by MikeHeiber** [comment]
@kthyng The Zenodo archive is now updated as requested, and your PR is merged into the joss-proofing branch.

**Comment by kthyng** [comment]
@whedon generate pdf from branch joss-proofing

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-proofing. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02307/joss.02307/10.21105.joss.02307.pdf)

**Comment by kthyng** [comment]
@whedon accept from branch joss-proofing

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1002/pssb.201350339 is OK
- 10.1088/1361-6633/80/2/026502 is OK
- 10.21105/joss.01072 is OK
- 10.1016/B978-0-08-102284-9.00010-3 is OK
- 10.21105/joss.01168 is OK
- 10.5281/zenodo.4008122 is OK
- 10.1002/aelm.201800821 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1707

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1707, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss-proofing 
```

**Comment by kthyng** [comment]
@whedon accept deposit=true from branch joss-proofing

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1301939718175764487) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1708
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02307](https://doi.org/10.21105/joss.02307)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congrats to @MikeHeiber on your new publication! Thanks to editor @pdebuyl and reviewers @mdavezac and @Luthaf. This wouldn't be possible without your time and expertise.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02307/status.svg)](https://doi.org/10.21105/joss.02307)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02307">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02307/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02307/status.svg
   :target: https://doi.org/10.21105/joss.02307
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02307/status.svg)](https://doi.org/10.21105/joss.02307)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
